include/lexy/dsl/context_counter.hpp

```diff
@@ -206,6 +206,54 @@ struct _ctx_ceq<H, T...> : branch_base
 
 namespace lexyd
 {
+template <int Value>
+struct _equals
+{
+    template <typename T>
+    constexpr bool operator()(T id) const noexcept
+    {
+        return id == Value;
+    }
+};
+template <int Value>
+using equals = _equals<Value>;
+
+template <int Value>
+struct _not_equals
+{
+    template <typename T>
+    constexpr bool operator()(T id) const noexcept
+    {
+        return id != Value;
+    }
+};
+template <int Value>
+using not_equals = _not_equals<Value>;
+
+template <int Value>
```

Comment:
Likewise, I would drop `_greater` and `_less` entirely. They're trivial to write if needed. (If there's demand, I'd rather expose them as additional member functions).